packages/utils/src/feature_toggle.rs

```diff
@@ -0,0 +1,414 @@
+        }
+
+        Ok(())
+    }
+
+    fn _resume<S: Storage>(storage: &mut S, features: Vec<String>) -> StdResult<()> {
+        for f in features {
+            Self::_set_feature_status(storage, f, FeatureStatus::Resumed)?;
+        }
+
+        Ok(())
+    }
+
+    fn _get_pauser<S: ReadonlyStorage>(storage: &S, key: HumanAddr) -> StdResult<Option<u8>> {
+        let feature_store =
+            ReadonlyBucket::multilevel(&[PREFIX_FEATURE_TOGGLE, PREFIX_PAUSERS], storage);
+        feature_store.may_load(key.0.as_bytes())
+    }
+
+    fn _set_pauser<S: Storage>(storage: &mut S, key: HumanAddr) -> StdResult<()> {
+        let mut feature_store =
+            Bucket::multilevel(&[PREFIX_FEATURE_TOGGLE, PREFIX_PAUSERS], storage);
+        feature_store.save(key.0.as_bytes(), &1_u8 /* value is insignificant */)
+    }
+
+    fn _remove_pauser<S: Storage>(storage: &mut S, key: HumanAddr) {
+        let mut feature_store: Bucket<S, u8> =
+            Bucket::multilevel(&[PREFIX_FEATURE_TOGGLE, PREFIX_PAUSERS], storage);
+        feature_store.remove(key.0.as_bytes())
+    }
+
+    fn _get_feature_status<S: ReadonlyStorage>(
+        storage: &S,
+        key: String,
+    ) -> StdResult<Option<FeatureStatus>> {
+        let feature_store =
+            ReadonlyBucket::multilevel(&[PREFIX_FEATURE_TOGGLE, PREFIX_FEATURES], storage);
+        feature_store.may_load(key.as_bytes())
+    }
+
+    fn _set_feature_status<S: Storage>(
+        storage: &mut S,
+        key: String,
+        item: FeatureStatus,
+    ) -> StdResult<()> {
+        let mut feature_store =
+            Bucket::multilevel(&[PREFIX_FEATURE_TOGGLE, PREFIX_FEATURES], storage);
+        feature_store.save(key.as_bytes(), &item)
+    }
+}
+
+#[derive(Serialize, Debug, Deserialize, Clone, JsonSchema, PartialEq)]
+pub enum FeatureStatus {
+    Resumed,
+    Stopped,
+}
+
+impl Default for FeatureStatus {
+    fn default() -> Self {
+        FeatureStatus::Resumed
+    }
+}
+
+#[derive(Serialize, Debug, Deserialize, Clone, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum FeatureToggleMsg {
+    Stop { features: Vec<String> },
+    Resume { features: Vec<String> },
+    SetPauser { address: HumanAddr },
+    RemovePauser { address: HumanAddr },
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+enum ResponseStatus {
+    Success,
+    Failure,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+enum HandleAnswer {
+    Stop { status: ResponseStatus },
+    Resume { status: ResponseStatus },
+    SetPauser { status: ResponseStatus },
+    RemovePauser { status: ResponseStatus },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum FeatureToggleQueryMsg {
+    Status {},
+    IsPauser { address: HumanAddr },
+}
```

Comment:
I'd add an example of using this type in the readme.

Also, `status` should include a list of features we want to query. An empty list then just do nothing.

Reply:
> `status` should include a list of features we want to query. An empty list then just do nothing

Good point, but regarding implementation - how would you implement the mapping between the querier's input to the potentially-enum based features?

Reply:
> `status` should include a list of features we want to query. An empty list then just do nothing

Good point, but regarding implementation - how would you implement the mapping between the querier's input to the potentially-enum based features?

Reply:
mmmm that is a very good question.... You could require that the enum also implements this trait:
https://doc.rust-lang.org/std/str/trait.FromStr.html
And then try to `.parse()` the user's strings into the enum?

Reply:
Actually, in `FeatureToggleMsg` and `FeatureToggleQueryMsg` you can just add a Generic parameter `<T: Deserialize>`, and for regular enums, it'll work just right without any boilerplate.